opslevel/resource_opslevel_check_package_version.go

```diff
@@ -0,0 +1,314 @@
+package opslevel
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+	"github.com/relvacode/iso8601"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &CheckPackageVersionResource{}
+	_ resource.ResourceWithImportState = &CheckPackageVersionResource{}
```

Comment:
Since this has a predicate we should add `_ resource.ResourceWithValidateConfig = & CheckPackageVersionResource{}` here 
and
`func (r * CheckPackageVersionResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) { ... }` down below.

[Here is a ValidateConfig example](https://github.com/OpsLevel/terraform-provider-opslevel/blob/main/opslevel/resource_opslevel_check_alert_source_usage.go#L169-L180), it lets devs know if the predicate in their Terraform config has issues at `terraform validate` time (pre-plan)